### ISSUE TITLE
Externalize xgboost learner from model json

### ISSUE THREAD

**Issue by rootfs** [sustainable-computing-io/kepler-model-server#189]
### What would you like to be added?

The current xgboost weight is embedded in `weight` as the following:

```json
 {
    "package": {
        "All_Weights": {
            "Bias_Weight": 0,
            "Categorical_Variables": {

            },
            "Numerical_Variables": {
                "bpf_cpu_time_us": {
                    "scale": 29170.333333333332,
                    "mean": 0,
                    "variance": 0,
                    "weight": {
                        "learner": {
				"gradient_booster": {
				}
			}
		}
	}
}
```

while the linear model uses `weight` in a different way:
```json
{
"All_Weights":

		{
		"Bias_Weight": 1.0,
		"Categorical_Variables": {"cpu_architecture": {"Sky Lake": {"weight": 1.0}}},
		"Numerical_Variables": {"cpu_cycles": {"mean": 0, "variance": 1.0, "weight": 1.0}}
		}
	}
```

It is hard to parse the right form of `weight` if it is overloaded. 

### Why is this needed?

Please add `xgboost_weight` to the same level of `All_Weights` to make it easy to unmarshal and remain backward compatible, e.g.

```json
 {
    "package": {
        "All_Weights": {},
        "XGboost_Weights": {
                  "learner": {
   		     "gradient_booster": {}
         }
    }
}
```

**Comment by sunya-ch** [comment]
How's about changing to following format?

```yaml
 {
    "package": {
        "All_Weights": {
		"Categorical_Variables":  <mapping from categorical variable to numerical value>,
		"Numerical_Variables": <mapping from numerical variable to scale value>,
                "XGboost_Weights": {
                          "learner": {
   		             "gradient_booster": {}
                 }
        }
    }
}
```
